filecoin-proofs/src/bin/paramfetch.rs

```diff
@@ -176,3 +300,42 @@ fn get_pending_parameter_ids(
         })
         .collect())
 }
+
+fn get_filenames_from_parameter_map(parameter_map: &ParameterMap) -> Result<Vec<String>> {
+    Ok(parameter_map.iter().map(|(k, _)| k.clone()).collect())
+}
+
+fn validate_parameter_file(parameter_map: &ParameterMap, filename: &str) -> Result<bool> {
+    let parameter_data = parameter_map_lookup(parameter_map, filename)?;
+    let digest = get_digest_for_file_within_cache(filename)?;
+
+    if parameter_data.digest != digest {
+        Ok(false)
+    } else {
+        Ok(true)
+    }
+}
+
+fn invalidate_parameter_file(filename: &str) -> Result<()> {
+    let parameter_file_path = get_full_path_for_file_within_cache(filename);
+    let target_parameter_file_path =
+        parameter_file_path.with_file_name(format!("{}-invalid-digest", filename));
+
+    if parameter_file_path.exists() {
+        rename(parameter_file_path, target_parameter_file_path)?;
+        Ok(())
+    } else {
+        Err(err_msg(ERROR_PARAMETER_FILE))
+    }
+}
+
+fn parameter_map_lookup<'a>(
+    parameter_map: &'a ParameterMap,
+    filename: &str,
+) -> Result<&'a ParameterData> {
```

Comment:
why not

```rust
impl ParameterMap {
  pub fn lookup(&self, filname: &str) -> Result<&ParameterData> {}
}
```

Reply:
No idea. It's code I pulled in from the original implementation.